src/timeline/app.js

```diff
@@ -26,9 +26,9 @@
  * along with OpenShot Library.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-
 // Initialize Angular application
 /*global App, timeline, angular*/
+var timeline = null;
 var App = angular.module("openshot-timeline", ["ui.bootstrap", "ngAnimate"]);
```

Comment:
Oh, hey, on the `App` global? _Apparently_, the way that's supposed to work is, `App` can be local to each source file:
```js
var App = angular.module("openshot-timeline");
```
...With that at the start of every JS source file _except_ `app.js`, the `App` variable doesn't have to be treated as a global. (The secret, I gather, is that while `angular.module(module_name, [requirements]);` _creates_ a new module, `angular.module(module_name);` just looks up a reference to the named module. Kind of like Qt's `QApplication()` singleton.)

Reply:
I tested the above, and it does seem to work fine and not cause any undue consternation to the JS code. Remains to be seen what Codacy will think of it, since we're really just replacing one undeclared global (`App`) with a different one (`angular`) in all of the directive sources.

Reply:
I tested the above, and it does seem to work fine and not cause any undue consternation to the JS code. Remains to be seen what Codacy will think of it, since we're really just replacing one undeclared global (`App`) with a different one (`angular`) in all of the directive sources.